### PR TITLE
Add a `fail_if_not_bottommost_level` to IngestExternalFileOptions

### PR DESCRIPTION
**Pull request by riversand963**
This new options allows application to specify that files must be
ingested to bottommost level, otherwise the ingestion will fail instead
of silently ingesting to a non-bottommost level.

Test plan:
make check

### CODE REVIEW

File: db/external_sst_file_ingestion_job.cc [hunk 1]
```diff
@@ -739,6 +739,9 @@ Status ExternalSstFileIngestionJob::AssignLevelAndSeqnoForIngestedFile(
   if (force_global_seqno) {
     *assigned_seqno = last_seqno + 1;
     if (compaction_style == kCompactionStyleUniversal || files_overlap_) {
+      if (ingestion_options_.fail_if_not_bottommost_level) {
+        return Status::InvalidArgument("Files cannot be ingested to Lmax");
```
Comment (ajkr) [inline db/external_sst_file_ingestion_job.cc:743]: Hm but the argument is valid. Maybe Incomplete or TryAgain? I actually like TryAgain assuming the error indicates the bottom-level range needs to be cleared.

Comment (ajkr) [reply db/external_sst_file_ingestion_job.cc:743]: Maybe TryAgain would make more sense coupled with an API comment on `fail_if_not_bottommost_level` that the user is responsible for ensuring the bottom-level is clear in the range. Anyways, up to you.

**Review by ajkr — APPROVED** [review]
LGTM!

**Comment by riversand963** [comment]
Thanks @ajkr for the review!

File: db/external_sst_file_ingestion_job.cc (hunk 1 above)
Comment (riversand963) [reply db/external_sst_file_ingestion_job.cc]: Done.

**Pull request closed without merging**